### PR TITLE
Simple Source traversal util

### PR DESCRIPTION
**Pull request by voznesenskym**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #103386
* #103456
* __->__ #103450
* #103445

lint

cc @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @msaroufim

### CODE REVIEW

File: test/dynamo/test_sources.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Owner(s): ["module: dynamo"]
+"""
+PYTEST_DONT_REWRITE (prevents pytest from rewriting assertions, which interferes
+with test_export_persist_assert)
+"""
```
Comment (ezyang) [inline test/dynamo/test_sources.py:5]: ??

File: test/dynamo/test_sources.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Owner(s): ["module: dynamo"]
```
Comment (ezyang) [inline test/dynamo/test_sources.py:1]: test_source.py for symmetry with the module name in question?

**Review by ezyang — APPROVED** [review]

File: test/dynamo/test_sources.py (hunk 2 above)
Comment (voznesenskym) [reply test/dynamo/test_sources.py:1]: I wanted to, but then I saw test_misc has MiscTests, test_functions has FunctionTests, so I kept it as is lol

**Pull request closed without merging**